### PR TITLE
Support schema version 2 of the Trivy JSON format.

### LINKED ISSUE #9711: Using Trivy as a source for Security warnings results in parse error
trivy version 0.55.2

We have followed the readthedocs and it still results in a parse error.

this is the command used to create the json 
trivy image --debug --offline-scan --input target/tar_images/${IMAGE_NAME}.tar -f json -o target/testresults/trivy/trivy-results.json

image
image

### PR DESCRIPTION
**Pull request by fniessink**
Closes #9711.

### CODE REVIEW

File: components/collector/src/source_collectors/trivy/security_warnings.py [hunk 1]
```diff
@@ -26,14 +26,24 @@ class TrivyJSONVulnerability(TypedDict):
...
     Target: str
     Vulnerabilities: list[TrivyJSONVulnerability] | None  # The examples in the Trivy docs show this key can be null
 
 
-TrivyJSON = list[TrivyJSONDependencyRepository]
+TriviJSONSchemaVersion1 = list[TrivyJSONResult]
+
+
+class TrivyJSONSchemaVersion2(TypedDict):
+    """Trivy JSON conform schema version 2, see https://github.com/aquasecurity/trivy/discussions/1050."""
```
Comment (wkoot) [inline components/collector/src/source_collectors/trivy/security_warnings.py]: Is this still accurate, does it exist? The [docs regarding container scanning](https://aquasecurity.github.io/trivy/v0.55/docs/target/container_image/) don't name a Trivy JSON format.
In fact, [section SBOM > Generation detail](https://aquasecurity.github.io/trivy/v0.55/docs/supply-chain/sbom/) states that CycloneDX or SPDX are the only two (SBOM) formats.
Note that Trivy version 0.55.2 was only released yesterday :D I think it'd be good to at least find docs/spec.
After digging through the docs for a while, it seems that the format is listed under [Reporting > JSON](https://aquasecurity.github.io/trivy/v0.55/docs/configuration/reporting/#json).
Should we really support this custom Trivy format, when it also offers the SARIF and SBOM formats?

File: components/collector/src/source_collectors/trivy/security_warnings.py [hunk 2]
```diff
@@ -46,9 +56,11 @@ class TrivyJSONSecurityWarnings(SecurityWarningsSourceCollector, JSONFileSourceC
     def _parse_json(self, json: JSON, filename: str) -> Entities:
         """Override to parse the vulnerabilities from the Trivy JSON."""
         entities = Entities()
-        for dependency_repository in cast(TrivyJSON, json):
-            target = dependency_repository["Target"]
-            for vulnerability in dependency_repository.get("Vulnerabilities") or []:
+        trivy_json = cast(TrivyJSON, json)
+        results = trivy_json["Results"] if isinstance(trivy_json, dict) else trivy_json
```
Comment (wkoot) [inline components/collector/src/source_collectors/trivy/security_warnings.py:67]: Pretty dirty that the `"Results"` section is only mentioned in the 3-year-old github discussion, but not on their official docs.

**Review by wkoot — APPROVED** [review]

File: components/collector/src/source_collectors/trivy/security_warnings.py (hunk 1 above)
Comment (fniessink) [reply components/collector/src/source_collectors/trivy/security_warnings.py]: I've opened a discussion at https://github.com/aquasecurity/trivy/discussions/7552. The response was to open a ticket to improve the docs, so apparently the format is still supported.

File: components/collector/src/source_collectors/trivy/security_warnings.py (hunk 2 above)
Comment (fniessink) [reply components/collector/src/source_collectors/trivy/security_warnings.py:67]: Yeah, but now at least there's a ticket to improve the docs :-). 

https://github.com/aquasecurity/trivy/issues/7553

**Pull request merged**